src/utils/api-client.js

```diff
@@ -1,6 +1,43 @@
 import {queryCache} from 'react-query'
 import * as auth from 'auth-provider'
-const apiURL = process.env.REACT_APP_API_URL
+import axios from 'axios';
+const apiURL = process.env.REACT_APP_API_URL;
+
+const axiosConfig = {
+  baseURL: apiURL,
+};
+
+const axiosClient = axios.create(axiosConfig);
+
+axiosClient.interceptors.request.use(
+  async (config) => {
+    const token = await auth.getToken();
+    const headers = {
+      Authorization: token ? `Bearer ${token}` : undefined,
+    };
+    return { ...config, headers: { ...config.headers, ...headers } };
+  },
+);
+
+axiosClient.interceptors.response.use(
+  async (response) => {
+    if (response.status === 401) {
+      await auth.logout();
+      // refresh the page for them
+      window.location.assign(window.location);
```

Comment:
Logout logic is not centralized here. We should only handle the auth.logout() in redux middleware, and only there.

Reply:
Thank you, refactored.

Reply:
Thank you, refactored.